osquery/tables/applications/vscode_extensions.cpp

```diff
@@ -116,14 +118,27 @@ QueryData genVSCodeExtensions(QueryContext& context) {
     if (directory == row.end() || uid == row.end()) {
       continue;
     }
+
+    // Add paths for both VSCode and VSCode Insiders
     confDirs.insert(
         {uid->second, fs::path(directory->second) / ".vscode-server"});
     confDirs.insert({uid->second, fs::path(directory->second) / ".vscode"});
+    confDirs.insert(
+        {uid->second, fs::path(directory->second) / ".vscode-server-insiders"});
+    confDirs.insert(
+        {uid->second, fs::path(directory->second) / ".vscode-insiders"});
   }
 
   for (const auto& confDir : confDirs) {
+    // Determine the version type
+    std::string version_type =
+        (confDir.second.string().find("insiders") != std::string::npos)
```

Comment:
Instead of making this generic check, I would just add the information we need in the `confDirs` set, so we have the correct information already there.

So store a struct with 3 values in it instead of the `std::pair`, with the third being the `version_type`.

And at this point I would also make `confDirs` -> `conf_dirs` to be consistent with the variable naming scheme.

Reply:
That makes sense, think that would be clearer too.

I didn't have time to make the change during the weekend but will do in the next couple days. Thank you for the feedback.

Reply:
Ok I made a `ConfDir` struct that contains the `uid`, `path` and `version_type` and am using it inside the `conf_dirs` set.

Let me know if this is not what you had in mind since my C++ is not great